tests/test_terrain.py

```diff
@@ -190,6 +193,208 @@ def test_attribute_functions_against_richdem(self, attribute: str) -> None:
+
+            if make_aspect:
+                # The aspect of RichDEM is returned in degrees, we convert to radians to match the others
+                terrain_attributes["aspect"] = np.deg2rad(_get_terrainattr_richdem(dem, attribute="aspect"))
+                # For flat slopes, RichDEM returns a 90° aspect by default, while GDAL return a 180° aspect
+                # We stay consistent with GDAL
+                slope_tmp = _get_terrainattr_richdem(dem, attribute="slope_radians")
+                terrain_attributes["aspect"][slope_tmp == 0] = np.pi
+
+            if make_hillshade:
+                # If a different z-factor was given, slopemap with exaggerated gradients.
+                if hillshade_z_factor != 1.0:
+                    slopemap = np.arctan(np.tan(terrain_attributes["slope"]) * hillshade_z_factor)
+                else:
+                    slopemap = terrain_attributes["slope"]
+
+                azimuth_rad = np.deg2rad(360 - hillshade_azimuth)
+                altitude_rad = np.deg2rad(hillshade_altitude)
+
+                # The operation below yielded the closest hillshade to GDAL (multiplying by 255 did not work)
+                # As 0 is generally no data for this uint8, we add 1 and then 0.5 for the rounding to occur between
+                # 1 and 255
+                terrain_attributes["hillshade"] = np.clip(
+                    1.5
+                    + 254
+                    * (
+                        np.sin(altitude_rad) * np.cos(slopemap)
+                        + np.cos(altitude_rad) * np.sin(slopemap) * np.sin(azimuth_rad - terrain_attributes["aspect"])
+                    ),
+                    0,
+                    255,
+                ).astype("float32")
+
+            if make_curvature:
+                terrain_attributes["curvature"] = _get_terrainattr_richdem(dem, attribute="curvature")
+
+            if make_planform_curvature:
+                terrain_attributes["planform_curvature"] = _get_terrainattr_richdem(dem, attribute="planform_curvature")
+
+            if make_profile_curvature:
+                terrain_attributes["profile_curvature"] = _get_terrainattr_richdem(dem, attribute="profile_curvature")
+
+            # Convert the unit if wanted.
+            if degrees:
+                for attr in ["slope", "aspect"]:
+                    if attr not in terrain_attributes:
+                        continue
+                    terrain_attributes[attr] = np.rad2deg(terrain_attributes[attr])
+
+            output_attributes = [terrain_attributes[key].reshape(dem.shape) for key in attribute]
+
+            if isinstance(dem, gu.Raster):
+                output_attributes = [
+                    gu.Raster.from_array(attr, transform=dem.transform, crs=dem.crs, nodata=-99999)
+                    for attr in output_attributes
+                ]
+
+            return output_attributes if len(output_attributes) > 1 else output_attributes[0]
+
```

Comment:
I understand that this code needs to be retrieved, but placing it in this test file makes it heavier and does not meet the needs of a test file. You're making good progress, and we could take advantage of that by having you look into using the `conftest.py` file with pytest. Otherwise, we could move this code into a file named `tests/richdem_utils.py`.